src/Documentation/DocumentationGenerator.php

```diff
@@ -348,13 +348,23 @@ public function generateFixerDocumentation(FixerInterface $fixer)
         return "{$doc}\n";
     }
 
-    private function generateSampleDiff(FixerInterface $fixer, CodeSampleInterface $sample, $sampleIndex, $ruleName)
+    private function generateSampleDiff(FixerInterface $fixer, CodeSampleInterface $sample, $sampleNumber, $ruleName)
     {
         if ($sample instanceof VersionSpecificCodeSampleInterface && !$sample->isSuitableFor(\PHP_VERSION_ID)) {
+            $existingFile = @file_get_contents($this->getFixerDocumentationFilePath($fixer));
+
+            if (false !== $existingFile) {
+                Preg::match("/\\RExample #{$sampleNumber}\\R.+?(?<diff>\\R\\.\\. code-block:: diff\\R\\R.*?)\\R(?:\\R\\S|$)/s", $existingFile, $matches);
```

Comment:
when running the `doc.php` file with your branch under PHP 7.4.3, I got the following result:
```
ker@dus:~/github/PHP-CS-Fixer λ git du
diff --git a/doc/rules/function_notation/phpdoc_to_return_type.rst b/doc/rules/function_notation/phpdoc_to_return_type.rst
index f7470c4e3..a3509c5a0 100644
--- a/doc/rules/function_notation/phpdoc_to_return_type.rst
+++ b/doc/rules/function_notation/phpdoc_to_return_type.rst
@@ -96,3 +96,12 @@ With configuration: ``['scalar_types' => false]``.
    +function foo(): Foo {}
     /** @return string */
     function bar() {}
+
+Example #5
+~~~~~~~~~~
+
+*Default* configuration.
+
+.. error::
+   Cannot generate diff for code sample #5 of rule phpdoc_to_return_type:
+   the sample is not suitable for current version of PHP (7.4.3).
```

do you also aim to cover that case?

Reply:
I don't see a solution for that right now: the code sample was never generated so we need to run the fixer on PHP 8.0 to be able to generate the diff for the first time. This PR will prevent the `doc.php` command from removing the code sample **after** it has been generated at least once on PHP 8.0.

Anyway if one is able to contribute changes that require PHP 8.0, they should be able to update the doc as well, so the remaining issue should not happen often.

Reply:
now, the code sample no5 is present on 2.15, but the doc.php generator is still trying to replace it with error message.

I found a reason for that, can you take a look, @julienfalque ?
https://github.com/julienfalque/PHP-CS-Fixer/pull/11

Reply:
Merged, thanks :)